src/ol/render/canvas/TextBuilder.js

```diff
@@ -41,6 +41,11 @@ export const TEXT_ALIGN = {
   'bottom': 1,
 };
 
+const defaultFillKey =
+  typeof defaultFillStyle == 'string'
+    ? defaultFillStyle
+    : '|' + getUid(defaultFillStyle);
+
```

Comment:
You know the type of the `defaultFillStyle`, and there is no way for the user to change it.

Reply:
Defensive coding.  Where `defaultFillStyle` is set anywhere else a key is obtained by the normal route, so any future change to the default would not break any code.

Reply:
I'd prefer a code base where we have tests rather than defensive coding.